src/gwt/src/org/rstudio/studio/client/rmarkdown/ui/RmdTemplateChooser.java

```diff
@@ -85,18 +86,24 @@ public void populateTemplates()
          @Override
          public void onResponseReceived(JsArray<RmdDocumentTemplate> templates)
          {
-            for (int i = 0; i < templates.length(); i++)
+            // Sort the list by package, then template name
+            JsArrayUtil.fillList(templates, templates_);
+            templates_.sort((RmdDocumentTemplate a, RmdDocumentTemplate b) ->
             {
-               final RmdDocumentTemplate template = templates.get(i);
+               return (a.getPackage() + a.getName()).compareTo(
```

Comment:
In theory this will sort in an unexpected way if one package name is a substring of the other; e.g.

```
foo + bar
fo + obaz
```

since "fo" < "foo", but "foobar" < "foobaz".

The chances of this happening in this context are unlikely but since ultimately we're dealing with user input I think it's worth comparing each component.

Reply:
Good point; updated to compare components individually.